src/config.rs

```diff
@@ -502,16 +502,14 @@ impl<'a> Cfg<'a> {
         &self,
         toolchain: Option<PartialToolchainDesc>,
     ) -> anyhow::Result<Toolchain<'_>> {
-        match toolchain {
+        let toolchain = match toolchain {
```

Comment:
`map().transpose()`?

Reply:
@djc I've made https://github.com/rust-lang/rustup/pull/3986 for this.